src/main/java/com/couchbase/lite/replicator/ChangeTracker.java

```diff
@@ -326,7 +326,8 @@ public void process(HttpRequest request, HttpContext context) throws HttpExcepti
                 Log.v(Log.TAG_CHANGE_TRACKER, "%s: Making request to %s", this, maskedRemoteWithoutCredentials);
                 HttpResponse response = httpClient.execute(request);
                 StatusLine status = response.getStatusLine();
-                if (status.getStatusCode() >= 300 && !Utils.isTransientError(status)) {
+                if ((status.getStatusCode() >= 300 && !Utils.isTransientError(status)) ||
+                        status.getStatusCode() >= 300 && mode != ChangeTrackerMode.LongPoll) {
```

Comment:
1. It is hard to compare with iOS ChangeTracker.m. .NET v1.2 is same with current version (without this change). https://github.com/couchbase/couchbase-lite-net/blob/release/1.2/src/Sharpened/Replicator/ChangeTracker.cs#L358
2. PullReplicator does not stop. ChangeTracker stops immediately if ChangeTracker is running with OneShot mode. (I need to investigate more relationship between PullReplicator continuous/oneshot mode and ChangeTracker's LongPoll/OneShot mode).